src/pcm.cpp

```diff
@@ -92,4 +94,104 @@ Result<PCM> PCM::fromMemory(std::span<std::byte const> wavBytes) {
 	}
 	return Error::eUnknown;
 }
+
+struct PCM::Streamer::File {
+	// one pinned optional per file type: can't use variant etc because can't move drwav
```

Comment:
This is definitely a problem: maybe we should encapsulate drwav with a smart pointer and disable copying for WAV & other file formats

Reply:
I'd say since it's internal and used only here, the current design is ok: technically `WAV` is not copiable/moveable, but C++17 onwards even such types can be "returned", so it won't prevent us from writing that kind of UB.

Reply:
I was considering whether to use unique_ptr or optional for each subtype, decided to go with the latter to avoid an extra heap alloc; this pimpl is already on the heap.